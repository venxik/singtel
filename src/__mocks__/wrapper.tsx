import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
