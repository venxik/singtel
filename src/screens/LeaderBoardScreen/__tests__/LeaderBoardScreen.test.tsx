import { screen } from '@testing-library/react-native';
import { renderWithProviders } from '../../../__mocks__';
import LeaderboardScreen from '../LeaderboardScreen';

describe('LeaderboardScreen', () => {
  it('match snapshot', async () => {
    renderWithProviders(<LeaderboardScreen />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('has correct item length', async () => {
    renderWithProviders(<LeaderboardScreen />, {
      preloadedState: {
        main: { leaderboards: [10], currentScore: 0, questionTracker: 0, selectedCategory: '' },
      },
    });
    const items = await screen.findAllByTestId('leaderboard-item');
    expect(items.length).toBe(1);
  });

  it('shows empty text', () => {
    renderWithProviders(<LeaderboardScreen />);
    expect(screen.getByTestId('text-empty')).toBeDefined();
  });
});
