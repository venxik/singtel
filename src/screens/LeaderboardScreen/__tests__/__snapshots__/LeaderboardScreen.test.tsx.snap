// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardScreen match snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      {
        "flex": 1,
        "paddingHorizontal": 20,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000",
              },
              [
                undefined,
                {
                  "fontFamily": "comic",
                },
              ],
            ]
          }
          testID="text-empty"
        >
          No data. Please play one round first!
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
